src/api/chassis/model/hDriveModel.cpp

```diff
@@ -144,6 +144,42 @@ void HDriveModel::arcade(const double iforwardSpeed, const double iyaw, const do
   middleMotor->moveVelocity(0);
 }
 
+void HDriveModel::curvature(const double iforwardSpeed, 
+                            const double icurvature, 
+                            const double ithreshold) {
+  // This code is adapted from WPIlib. All credit goes to them. Link:
+  // https://github.com/wpilibsuite/allwpilib/blob/main/wpilibc/src/main/native/cpp/drive/DifferentialDrive.cpp#L49
+  double forwardSpeed = std::clamp(iforwardSpeed, -1.0, 1.0);
+  if (std::abs(forwardSpeed) < ithreshold) {
+    forwardSpeed = 0;
+  }
+
+  double curvature = std::clamp(icurvature, -1.0, 1.0);
+  if (std::abs(curvature) < ithreshold) {
+    curvature = 0;
+  }
+
+  // the algorithm switches to arcade when forward speed is 0 to allow point turns.
+  if(forwardSpeed == 0){
+    arcade(forwardSpeed, curvature, ithreshold);
+    return;
+  }
+
+  double leftSpeed = forwardSpeed + std::abs(forwardSpeed) * curvature;
+  double rightSpeed = forwardSpeed - std::abs(forwardSpeed) * curvature;
+  double maxSpeed = std::max(leftSpeed, rightSpeed);
+
+  // normalizes output
+  if(maxSpeed > 1.0){
+    leftSpeed /= maxSpeed;
+	  rightSpeed /= maxSpeed;
+  }
+
+  leftSideMotor->moveVoltage(static_cast<int16_t>(leftSpeed * maxVoltage));
+  rightSideMotor->moveVoltage(static_cast<int16_t>(rightSpeed * maxVoltage));
+  middleMotor->moveVelocity(0);
```

Comment:
I'm not entirely sure how often this would be used, but for consistency's sake it might be nice to add an `hCurvature` function like how `arcade` has a `hArcade` companion.

Reply:
I have added the hCurvature() implementation in the latest commit. Thank you.

I was also thinking about whether to implement an xCurvature() function for the sake of consistency. However, I don't think scaling the (independent) rotational output with the forward output would make much sense for a x drive. I know that @Yessir120 has a x drive on hand right now, and maybe he can help test if the algorithm would make any sense.

Reply:
> I don't think scaling the (independent) rotational output with the forward output would make much sense for a x drive

Agree. I don't think it would make sense, especially when strafing gets added.